Sources/Screens/DeveloperMenu/DeveloperMenuViewController.swift

```diff
@@ -1,16 +1,121 @@
+                formatResultLabel(label: happyPathResultLabel,
+                                  text: "Success: \(String(data: data!, encoding: .utf8) ?? "no body")",
+                                  textColor: .gdsGreen)
+            } catch let error as ServerError {
+                formatResultLabel(label: happyPathResultLabel,
+                                  text: "Error code: \(error.errorCode)\nEndpoint: \(error.endpoint ?? "missing")",
+                                  textColor: .red)
+            } catch {
+                formatResultLabel(label: happyPathResultLabel, text: "Error", textColor: .red)
+            }
+            happyPathButton.isLoading = false
+        }
+    }
+    
+    @IBOutlet private var happyPathResultLabel: UILabel! {
+        didSet {
+            happyPathResultLabel.isHidden = true
+            happyPathResultLabel.accessibilityIdentifier = "sts-happy-path-result"
+        }
+    }
+    
+    @IBOutlet private var unhappyPathButton: RoundedButton! {
+        didSet {
+            if AppEnvironment.callingSTSEnabled {
+                unhappyPathButton.titleLabel?.adjustsFontForContentSizeCategory = true
+                unhappyPathButton.setTitle("Hello World Error", for: .normal)
+            } else {
+                unhappyPathButton.isHidden = true
+            }
+            unhappyPathButton.accessibilityIdentifier = "sts-unhappy-path-button"
+        }
+    }
+    
+    @IBAction private func unhappyPathButtonAction(_ sender: Any) {
+        unhappyPathButton.isLoading = true
+        helloWorldUnhappyPath()
+    }
+    
+    private func helloWorldUnhappyPath() {
+        Task {
+            do {
+                _ = try await networkClient?.makeAuthorizedRequest(exchangeRequest: URLRequest(url: AppEnvironment.stsToken),
+                                                                   scope: "sts-test.hello-world",
+                                                                   request: URLRequest(url: AppEnvironment.stsHelloWorld))
+            } catch let error as ServerError {
+                formatResultLabel(label: unhappyPathResultLabel,
+                                  text: "Error code: \(error.errorCode)\nEndpoint: \(error.endpoint ?? "missing")",
+                                  textColor: .red)
+            } catch {
+                formatResultLabel(label: unhappyPathResultLabel,
+                                  text: "Error",
+                                  textColor: .red)
+            }
+            unhappyPathButton.isLoading = false
+        }
+    }
+    
+    @IBOutlet private var unhappyPathResultLabel: UILabel! {
+        didSet {
+            unhappyPathResultLabel.isHidden = true
+            unhappyPathResultLabel.accessibilityIdentifier = "sts-unhappy-path-result"
+        }
+    }
+
+    private func formatResultLabel(label: UILabel,
```

Comment:
Because passing an instance variable into a function always feels wrong to me, I might suggest this solution instead:

```
fileprivate extension UILabel {
    func showErrorMessage(_ error: ServerError? = nil) {
        textColor = .red
        isHidden = false
        if let error {
            self.text = "Error code: \(error.errorCode)\nEndpoint: \(error.endpoint ?? "missing")"
        } else {
            self.text = "Error"
        }
    }
    
    func showSuccessMessage(_ message: String) {
        self.textColor = .gdsGreen
        isHidden = false
        text = message
    }
}
```
Note I am not setting font here.  I would do that in the didSet of the labels themselves, as I don't see a reason we would change this based on response.

Reply:
Good shout, thanks! I've not used fileprivate much, but in this case is it best to include this in the ViewController or make a file as an extension of UILabel (like what we have for String and LAContext)?